### DISCUSSION TITLE
Feature Request: Multi-User websites

### DISCUSSION THREAD

**Discussion by patrykgz** [umami-software/umami#907] [Ideas]
Hi all, would it be possible to implement shared access to analytics data through multiple accounts in Umami? For example, let's say our instance has the following:

* Website A
* Website B
* User C
* User D

I would like to modify the existing permission implementation to support User C AND User D having access to Website A, but only User C can access Website B.

Looking at the db schema Umami uses, this does not seem like a quick fix and is probably a breaking change. As I understand the schema currently, Umami has a column which denotes the user id that has access to a website analytics data. My suggested change is to add a new table to the schema which has the access rights per user and website. For example:

| UserId | WebsiteId |
| ----------- | --------------- |

Or perhaps the following

| UserId | WebsiteId | Role | 
| ----------- | --------------- | ------- |

Thanks :)

**Comment by mikecao** [comment]
This has been discussed before #118 , #434. But basically, you don't need a complex role/permission setup. If you want to share stats with someone, send them a share URL. Being an owner of a website means you have the right to delete it. I think most people don't want that behavior when sharing stats.

**Reply by t3brightside** [reply]
I think user rights handling is a must in bigger use cases. Share URL is not an option to work with multiple sites data daily especially if the data should not be public.

Lets say...
If a person has to work with 5 different websites daily and these websites all have different owner accounts. But there's more websites in the same system still. Besides that user is not the website owner but just an analyst.
That means data should be all read only and gathered under one user account.

There's a lot of use cases one could come up with. But I would say user rights handling it's a must as soon as it's used on bigger companies or projects that involve more than one person working on the data.

Right now what I see happening that website owners start to share account data, means sharing passwords. And we all know sharing user accounts is a big NO-NO.

Where I would start (I know easier said than done!):
- enabling read only on/off for users (read only - no edit, reset, and delete)
- assigning websites to user accounts by admin

**Reply by Amerlander** [reply]
How about an option to import share URLs and display them like the owned pages in the statistics. That could also enable a federated system, when you are able to import URLs from other umami installations.

**Reply by MagicLike** [reply]
I have also made a proposal here: #1131

**Comment by MostafaaRamadan** [comment]
@mikecao Is there any updates regarding this feature ?
